rest-api/src/main/kotlin/fund/cyber/markets/rest/handler/HistoHandler.kt

```diff
@@ -0,0 +1,77 @@
+package fund.cyber.markets.rest.handler
+
+import fund.cyber.markets.common.booleanValue
+import fund.cyber.markets.common.intValue
+import fund.cyber.markets.common.longValue
+import fund.cyber.markets.common.stringValue
+import fund.cyber.markets.dao.service.TickerDaoService
+import fund.cyber.markets.rest.configuration.AppContext
+import fund.cyber.markets.rest.model.ConversionType
+import fund.cyber.markets.rest.model.HistoEntity
+import fund.cyber.markets.rest.model.TickerData
+import io.undertow.server.HttpHandler
+import io.undertow.server.HttpServerExchange
+
+class HistoHandler(
+    private val duration: Long,
+    private val tickerDaoService: TickerDaoService = AppContext.tickerDaoService
+) : AbstractHandler(), HttpHandler {
+
+    override fun handleRequest(httpExchange: HttpServerExchange) {
+
+        val params = httpExchange.queryParameters
+        val base = params["fsym"]?.stringValue()
+        val quote = params["tsym"]?.stringValue()
+        var exchange = params["e"]?.stringValue()
+        var tryConversion = params["tryConversion"]?.booleanValue()
+        var limit = params["limit"]?.intValue()
+        var timestamp = params["toTs"]?.longValue()
+
+        if (base == null || quote == null || base == quote) {
+            handleBadRequest("Bad parameters", httpExchange)
+        }
+        if (exchange == null) {
```

Comment:
can be done int dedfinition line via  ':? "ALL"'
same for below items 
[example](https://github.com/cyberFund/cyber-search/blob/master/search-api/src/main/kotlin/fund/cyber/search/handler/SearchHandler.kt)

Reply:
done